.github/workflows/ci.yml

```diff
@@ -1,31 +1,41 @@
 name: ci
 on:
   push:
-   branches:
+    branches:
       - master
   pull_request:
 jobs:
-  build:
+  build-and-test:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node:
-          - 14
-          - 16
-          - 18
+        node: [14, 16, 18]
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node }}
       - run: npm install
       - run: npm run test:ci
+  integration-tests:
+    runs-on: ubuntu-latest
+    env:
+      AUTH0_DOMAIN: ${{ secrets.AUTH0_DOMAIN }}
+      AUTH0_CLIENT_ID: ${{ secrets.AUTH0_CLIENT_ID }}
+      AUTH0_CLIENT_SECRET: ${{ secrets.AUTH0_CLIENT_SECRET }}
+      AUTH0_API_AUDIENCE: ${{ secrets.AUTH0_API_AUDIENCE }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: .nvmrc
+      - run: npm install
+      - run: npm run test:integration
```

Comment:
run the integration tests in the previous step

Reply:
The original intention was to run integration tests once on a given Node version. We want to make sure we integrate properly with 3rd party services and I don't expect different outcomes based on the Node version.

We are about to have a matrix of 4 Node versions and this would mean a 4x traffic against our Auth0 free tier plan.

Btw, happy to move all into a single `build` job, if the case.

Reply:
fair point but I think it's valuable to run integration tests against the matrix